cloud/gcp: TestKMSAssumeRoleGCP failed

cloud/gcp.TestKMSAssumeRoleGCP [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21244325?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21244325?buildTab=artifacts#/) on master @ [4873a884da297307ecaf032119a6ad3336234446](https://github.com/cockroachdb/cockroach/commits/4873a884da297307ecaf032119a6ad3336234446):


```
        net.(*conn).Read(0xc0000fdd70, {0xc00d214000?, 0x7fbfb66cb388?, 0x7fbfff3bdf30?})
        	GOROOT/src/net/net.go:196 +0x45
        crypto/tls.(*atLeastReader).Read(0xc0156aa2a0, {0xc00d214000?, 0x17fb?, 0x0?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc0197c5428, {0x79fa280, 0xc0156aa2a0})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc0197c5188, {0x79f5ec0, 0xc0000fdd70}, 0x45b694?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc0197c5188, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc0197c5188, {0xc00d362000, 0x8000, 0x30?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        bufio.(*Reader).Read(0xc010a91b60, {0xc003b22664, 0x9, 0xc00296c801?})
        	GOROOT/src/bufio/bufio.go:245 +0x197
        io.ReadAtLeast({0x79f4180, 0xc010a91b60}, {0xc003b22664, 0x9, 0x9}, 0x9)
        	GOROOT/src/io/io.go:335 +0x8e
        io.ReadFull(...)
        	GOROOT/src/io/io.go:354
        golang.org/x/net/http2.readFrameHeader({0xc003b22664, 0x9, 0x0?}, {0x79f4180?, 0xc010a91b60?})
        	external/org_golang_x_net/http2/frame.go:242 +0x65
        golang.org/x/net/http2.(*Framer).ReadFrameHeader(0xc003b22620)
        	external/org_golang_x_net/http2/frame.go:505 +0x6b
        golang.org/x/net/http2.(*Framer).ReadFrame(0xc003b22620)
        	external/org_golang_x_net/http2/frame.go:564 +0x18
        google.golang.org/grpc/internal/transport.(*http2Client).reader(0xc012b2ab48, 0xc00e823b20)
        	external/org_golang_google_grpc/internal/transport/http2_client.go:1595 +0x226
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 50087
        	external/org_golang_google_grpc/internal/transport/http2_client.go:397 +0x1c9e
        Leaked goroutine: goroutine 50075 [select]:
        google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc00decae60, 0x1)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:418 +0x10d
        google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc00e823b90)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:552 +0x78
        google.golang.org/grpc/internal/transport.newHTTP2Client.func6()
        	external/org_golang_google_grpc/internal/transport/http2_client.go:451 +0x85
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 50087
        	external/org_golang_google_grpc/internal/transport/http2_client.go:449 +0x223b
        Leaked goroutine: goroutine 50085 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc00e6be640, {0x7a4fce0, 0xc00e5e0690})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x109
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 50037
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x11a
        Leaked goroutine: goroutine 50086 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc00e6be6a0, {0x7a4fce0, 0xc00e5e06e0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x109
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 50037
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x11a
--- FAIL: TestKMSAssumeRoleGCP (6.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKMSAssumeRoleGCP.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-61456

cloud/gcp.TestKMSAssumeRoleGCP [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21251520?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21251520?buildTab=artifacts#/) on master @ [84427ff3424990c1c6452473a73c9a0d5aa46ec9](https://github.com/cockroachdb/cockroach/commits/84427ff3424990c1c6452473a73c9a0d5aa46ec9):


```
        internal/poll.runtime_pollWait(0x7132d7081e00, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc022f73b80?, 0xc01914e000?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc022f73b80, {0xc01914e000, 0x1800, 0x1800})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc022f73b80, {0xc01914e000?, 0xc01914e000?, 0x5?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc00f1f47c8, {0xc01914e000?, 0x7132dc479348?, 0x7133276e95c0?})
        	GOROOT/src/net/net.go:196 +0x45
        crypto/tls.(*atLeastReader).Read(0xc00efa8e40, {0xc01914e000?, 0x17fb?, 0x0?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc01e74e628, {0x7a18dc0, 0xc00efa8e40})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc01e74e388, {0x7a14a00, 0xc00f1f47c8}, 0x45b694?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc01e74e388, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc01e74e388, {0xc002b36000, 0x8000, 0x30?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        bufio.(*Reader).Read(0xc018e81980, {0xc0185c6ac4, 0x9, 0xc0180c3101?})
        	GOROOT/src/bufio/bufio.go:245 +0x197
        io.ReadAtLeast({0x7a12cc0, 0xc018e81980}, {0xc0185c6ac4, 0x9, 0x9}, 0x9)
        	GOROOT/src/io/io.go:335 +0x8e
        io.ReadFull(...)
        	GOROOT/src/io/io.go:354
        golang.org/x/net/http2.readFrameHeader({0xc0185c6ac4, 0x9, 0x0?}, {0x7a12cc0?, 0xc018e81980?})
        	external/org_golang_x_net/http2/frame.go:242 +0x65
        golang.org/x/net/http2.(*Framer).ReadFrameHeader(0xc0185c6a80)
        	external/org_golang_x_net/http2/frame.go:505 +0x6b
        golang.org/x/net/http2.(*Framer).ReadFrame(0xc0185c6a80)
        	external/org_golang_x_net/http2/frame.go:564 +0x18
        google.golang.org/grpc/internal/transport.(*http2Client).reader(0xc00d170b48, 0xc00dad04d0)
        	external/org_golang_google_grpc/internal/transport/http2_client.go:1595 +0x226
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 60217
        	external/org_golang_google_grpc/internal/transport/http2_client.go:397 +0x1c9e
        Leaked goroutine: goroutine 60232 [select]:
        google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc014c58eb0, 0x1)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:418 +0x10d
        google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc00dad0540)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:552 +0x78
        google.golang.org/grpc/internal/transport.newHTTP2Client.func6()
        	external/org_golang_google_grpc/internal/transport/http2_client.go:451 +0x85
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 60217
        	external/org_golang_google_grpc/internal/transport/http2_client.go:449 +0x223b
--- FAIL: TestKMSAssumeRoleGCP (6.79s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKMSAssumeRoleGCP.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

cloud/gcp.TestKMSAssumeRoleGCP [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21259706?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21259706?buildTab=artifacts#/) on master @ [a3ec6a49f1829d4aa8f00c580331ac0e5e42fa1c](https://github.com/cockroachdb/cockroach/commits/a3ec6a49f1829d4aa8f00c580331ac0e5e42fa1c):


```
        internal/poll.runtime_pollWait(0x734a9d36b400, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc017ed8500?, 0xc00a594000?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc017ed8500, {0xc00a594000, 0x1800, 0x1800})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc017ed8500, {0xc00a594000?, 0xc00a59403e?, 0x5?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc01e479f48, {0xc00a594000?, 0x734a963c6fe0?, 0x734ae4767f30?})
        	GOROOT/src/net/net.go:196 +0x45
        crypto/tls.(*atLeastReader).Read(0xc01791e978, {0xc00a594000?, 0x17bd?, 0x0?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc015f849a8, {0x7a23900, 0xc01791e978})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc015f84708, {0x7a1f540, 0xc01e479f48}, 0x45b694?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc015f84708, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc015f84708, {0xc00a5ac000, 0x8000, 0x30?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        bufio.(*Reader).Read(0xc016561020, {0xc0029c2044, 0x9, 0xc01ee90b01?})
        	GOROOT/src/bufio/bufio.go:245 +0x197
        io.ReadAtLeast({0x7a1d800, 0xc016561020}, {0xc0029c2044, 0x9, 0x9}, 0x9)
        	GOROOT/src/io/io.go:335 +0x8e
        io.ReadFull(...)
        	GOROOT/src/io/io.go:354
        golang.org/x/net/http2.readFrameHeader({0xc0029c2044, 0x9, 0x0?}, {0x7a1d800?, 0xc016561020?})
        	external/org_golang_x_net/http2/frame.go:242 +0x65
        golang.org/x/net/http2.(*Framer).ReadFrameHeader(0xc0029c2000)
        	external/org_golang_x_net/http2/frame.go:505 +0x6b
        golang.org/x/net/http2.(*Framer).ReadFrame(0xc0029c2000)
        	external/org_golang_x_net/http2/frame.go:564 +0x18
        google.golang.org/grpc/internal/transport.(*http2Client).reader(0xc014ade008, 0xc00d62a930)
        	external/org_golang_google_grpc/internal/transport/http2_client.go:1595 +0x226
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 59057
        	external/org_golang_google_grpc/internal/transport/http2_client.go:397 +0x1c9e
        Leaked goroutine: goroutine 59082 [select]:
        google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc013e80c80, 0x1)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:418 +0x10d
        google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc00d62aa10)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:552 +0x78
        google.golang.org/grpc/internal/transport.newHTTP2Client.func6()
        	external/org_golang_google_grpc/internal/transport/http2_client.go:451 +0x85
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 59057
        	external/org_golang_google_grpc/internal/transport/http2_client.go:449 +0x223b
--- FAIL: TestKMSAssumeRoleGCP (6.56s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKMSAssumeRoleGCP.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

cloud/gcp.TestKMSAssumeRoleGCP [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21268265?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21268265?buildTab=artifacts#/) on master @ [c61065f38befa385fdb8061a3bf15bd46d709730](https://github.com/cockroachdb/cockroach/commits/c61065f38befa385fdb8061a3bf15bd46d709730):


```
        internal/poll.runtime_pollWait(0x71bcb2a08200, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc020481b80?, 0xc00d807800?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc020481b80, {0xc00d807800, 0x1800, 0x1800})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc020481b80, {0xc00d807800?, 0xc00d807800?, 0x5?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc001e2f058, {0xc00d807800?, 0x71bcb40fe518?, 0x71bd023705c0?})
        	GOROOT/src/net/net.go:196 +0x45
        crypto/tls.(*atLeastReader).Read(0xc0160603a8, {0xc00d807800?, 0x17fb?, 0x0?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc0112df428, {0x7a40bc0, 0xc0160603a8})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc0112df188, {0x7a3c800, 0xc001e2f058}, 0x45b694?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc0112df188, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc0112df188, {0xc002c3c000, 0x8000, 0x30?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        bufio.(*Reader).Read(0xc014fbe960, {0xc012bd6204, 0x9, 0xc001e1d701?})
        	GOROOT/src/bufio/bufio.go:245 +0x197
        io.ReadAtLeast({0x7a3aac0, 0xc014fbe960}, {0xc012bd6204, 0x9, 0x9}, 0x9)
        	GOROOT/src/io/io.go:335 +0x8e
        io.ReadFull(...)
        	GOROOT/src/io/io.go:354
        golang.org/x/net/http2.readFrameHeader({0xc012bd6204, 0x9, 0x0?}, {0x7a3aac0?, 0xc014fbe960?})
        	external/org_golang_x_net/http2/frame.go:242 +0x65
        golang.org/x/net/http2.(*Framer).ReadFrameHeader(0xc012bd61c0)
        	external/org_golang_x_net/http2/frame.go:505 +0x6b
        golang.org/x/net/http2.(*Framer).ReadFrame(0xc012bd61c0)
        	external/org_golang_x_net/http2/frame.go:564 +0x18
        google.golang.org/grpc/internal/transport.(*http2Client).reader(0xc01bdca008, 0xc01e112ee0)
        	external/org_golang_google_grpc/internal/transport/http2_client.go:1595 +0x226
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 53083
        	external/org_golang_google_grpc/internal/transport/http2_client.go:397 +0x1c9e
        Leaked goroutine: goroutine 53092 [select]:
        google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc017aa0050, 0x1)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:418 +0x10d
        google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc01e112f50)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:552 +0x78
        google.golang.org/grpc/internal/transport.newHTTP2Client.func6()
        	external/org_golang_google_grpc/internal/transport/http2_client.go:451 +0x85
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 53083
        	external/org_golang_google_grpc/internal/transport/http2_client.go:449 +0x223b
--- FAIL: TestKMSAssumeRoleGCP (6.88s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKMSAssumeRoleGCP.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

cloud/gcp.TestKMSAssumeRoleGCP [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21273051?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21273051?buildTab=artifacts#/) on master @ [d608df0444b8838419b12c7873750bd120115900](https://github.com/cockroachdb/cockroach/commits/d608df0444b8838419b12c7873750bd120115900):


```
        internal/poll.runtime_pollWait(0x720d5aefb800, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc012440c80?, 0xc002b98800?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc012440c80, {0xc002b98800, 0x1800, 0x1800})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc012440c80, {0xc002b98800?, 0xc002b98800?, 0x5?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc015303dd0, {0xc002b98800?, 0x720d5c570930?, 0x720da6d43108?})
        	GOROOT/src/net/net.go:196 +0x45
        crypto/tls.(*atLeastReader).Read(0xc001210558, {0xc002b98800?, 0x17fb?, 0x0?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc0124769a8, {0x7a46280, 0xc001210558})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc012476708, {0x7a41ec0, 0xc015303dd0}, 0x45b694?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc012476708, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc012476708, {0xc00f662000, 0x8000, 0x30?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        bufio.(*Reader).Read(0xc010bff320, {0xc0154be204, 0x9, 0xc01e4aa601?})
        	GOROOT/src/bufio/bufio.go:245 +0x197
        io.ReadAtLeast({0x7a40180, 0xc010bff320}, {0xc0154be204, 0x9, 0x9}, 0x9)
        	GOROOT/src/io/io.go:335 +0x8e
        io.ReadFull(...)
        	GOROOT/src/io/io.go:354
        golang.org/x/net/http2.readFrameHeader({0xc0154be204, 0x9, 0x0?}, {0x7a40180?, 0xc010bff320?})
        	external/org_golang_x_net/http2/frame.go:242 +0x65
        golang.org/x/net/http2.(*Framer).ReadFrameHeader(0xc0154be1c0)
        	external/org_golang_x_net/http2/frame.go:505 +0x6b
        golang.org/x/net/http2.(*Framer).ReadFrame(0xc0154be1c0)
        	external/org_golang_x_net/http2/frame.go:564 +0x18
        google.golang.org/grpc/internal/transport.(*http2Client).reader(0xc0149fb208, 0xc0012c4850)
        	external/org_golang_google_grpc/internal/transport/http2_client.go:1595 +0x226
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 42672
        	external/org_golang_google_grpc/internal/transport/http2_client.go:397 +0x1c9e
        Leaked goroutine: goroutine 42740 [select]:
        google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc017d28690, 0x1)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:418 +0x10d
        google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc0012c4930)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:552 +0x78
        google.golang.org/grpc/internal/transport.newHTTP2Client.func6()
        	external/org_golang_google_grpc/internal/transport/http2_client.go:451 +0x85
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 42672
        	external/org_golang_google_grpc/internal/transport/http2_client.go:449 +0x223b
--- FAIL: TestKMSAssumeRoleGCP (6.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKMSAssumeRoleGCP.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

cloud/gcp.TestKMSAssumeRoleGCP [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21279637?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21279637?buildTab=artifacts#/) on master @ [1225cbf40c0decb285c22eefd4e11096a34932a3](https://github.com/cockroachdb/cockroach/commits/1225cbf40c0decb285c22eefd4e11096a34932a3):


```
        internal/poll.runtime_pollWait(0x7695df0aa400, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc00f566400?, 0xc011048800?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc00f566400, {0xc011048800, 0x1800, 0x1800})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc00f566400, {0xc011048800?, 0xc011048800?, 0x5?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc020a183f8, {0xc011048800?, 0x7695e6cae630?, 0x76963337bf30?})
        	GOROOT/src/net/net.go:196 +0x45
        crypto/tls.(*atLeastReader).Read(0xc01a29c810, {0xc011048800?, 0x17fb?, 0x0?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc00f20db28, {0x7a50bc0, 0xc01a29c810})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc00f20d888, {0x7a4c800, 0xc020a183f8}, 0x45b694?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc00f20d888, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc00f20d888, {0xc00bee6000, 0x8000, 0x30?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        bufio.(*Reader).Read(0xc01851f6e0, {0xc01eccdb64, 0x9, 0xc002565601?})
        	GOROOT/src/bufio/bufio.go:245 +0x197
        io.ReadAtLeast({0x7a4aac0, 0xc01851f6e0}, {0xc01eccdb64, 0x9, 0x9}, 0x9)
        	GOROOT/src/io/io.go:335 +0x8e
        io.ReadFull(...)
        	GOROOT/src/io/io.go:354
        golang.org/x/net/http2.readFrameHeader({0xc01eccdb64, 0x9, 0x0?}, {0x7a4aac0?, 0xc01851f6e0?})
        	external/org_golang_x_net/http2/frame.go:242 +0x65
        golang.org/x/net/http2.(*Framer).ReadFrameHeader(0xc01eccdb20)
        	external/org_golang_x_net/http2/frame.go:505 +0x6b
        golang.org/x/net/http2.(*Framer).ReadFrame(0xc01eccdb20)
        	external/org_golang_x_net/http2/frame.go:564 +0x18
        google.golang.org/grpc/internal/transport.(*http2Client).reader(0xc019c6fd48, 0xc01f11c690)
        	external/org_golang_google_grpc/internal/transport/http2_client.go:1595 +0x226
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 46834
        	external/org_golang_google_grpc/internal/transport/http2_client.go:397 +0x1c9e
        Leaked goroutine: goroutine 46836 [select]:
        google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc00ec638b0, 0x1)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:418 +0x10d
        google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc01f11c700)
        	external/org_golang_google_grpc/internal/transport/controlbuf.go:552 +0x78
        google.golang.org/grpc/internal/transport.newHTTP2Client.func6()
        	external/org_golang_google_grpc/internal/transport/http2_client.go:451 +0x85
        created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 46834
        	external/org_golang_google_grpc/internal/transport/http2_client.go:449 +0x223b
--- FAIL: TestKMSAssumeRoleGCP (6.63s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKMSAssumeRoleGCP.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cloud/kms_test_utils.go
+++ b/pkg/cloud/kms_test_utils.go
@@ -53,6 +53,9 @@ func KMSEncryptDecrypt(t *testing.T, kmsURI string, env KMSEnv) {
 	ctx := context.Background()
 	kms, err := KMSFromURI(ctx, kmsURI, env)
 	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, kms.Close())
+	}()
 
 	t.Run("simple encrypt decrypt", func(t *testing.T) {
 		sampleBytes := "hello world"
@@ -64,8 +67,6 @@ func KMSEncryptDecrypt(t *testing.T, kmsURI string, env KMSEnv) {
 		require.NoError(t, err)
 
 		require.True(t, bytes.Equal(decryptedBytes, []byte(sampleBytes)))
-
-		require.NoError(t, kms.Close())
 	})
 }
 
@@ -75,6 +76,9 @@ func CheckNoKMSAccess(t *testing.T, kmsURI string, env KMSEnv) {
 	ctx := context.Background()
 	kms, err := KMSFromURI(ctx, kmsURI, env)
 	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, kms.Close())
+	}()
 
 	_, err = kms.Encrypt(ctx, []byte("test bytes"))
 	if err == nil {

